src/stream/stream/max.rs

```diff
@@ -11,29 +10,23 @@ use crate::task::{Context, Poll};
 pin_project! {
     #[doc(hidden)]
     #[allow(missing_debug_implementations)]
-    pub struct MaxFuture<S, F, T> {
+    pub struct MaxFuture<S, T> {
```

Comment:
this seems unrelated, why this change?

Reply:
I think it is unnecessary type argument. 
Certainly that may be different from what  I'm doing with this PR.